src/planner/binder/tableref/bind_basetableref.cpp

```diff
@@ -234,17 +234,21 @@ unique_ptr<BoundTableRef> Binder::Bind(BaseTableRef &ref) {
 	auto table_or_view =
 	    entry_retriever.GetEntry(ref.catalog_name, ref.schema_name, table_lookup, OnEntryNotFound::RETURN_NULL);
 	// we still didn't find the table
-	if (GetBindingMode() == BindingMode::EXTRACT_NAMES) {
+	if (GetBindingMode() == BindingMode::EXTRACT_NAMES || GetBindingMode() == BindingMode::EXTRACT_QUALIFIED_NAMES) {
 		if (!table_or_view || table_or_view->type == CatalogType::TABLE_ENTRY) {
-			// if we are in EXTRACT_NAMES, we create a dummy table ref
-			AddTableName(ref.table_name);
+			// if we are in EXTRACT_NAMES or EXTRACT_QUALIFIED_NAMES, we create a dummy table ref
+			if (GetBindingMode() == BindingMode::EXTRACT_QUALIFIED_NAMES) {
+				AddTableName(ref.ToString());
+			} else {
+				AddTableName(ref.table_name);
```

Comment:
I'm not sure but if you'd call `GetTableNames` with `BindingMode::EXTRACT_QUALIFIED_NAMES` you might get the table name escaped even though it wouldn't be if it was called with `BindingMode::EXTRACT_NAMES` for the same table? If that's true - I scanned quickly but don't see `BaseTableRef.table_name` getting escaped anywhere - then maybe we want to document that? I guess there's not really any way around it and my feeling is that it's not bad, as long as it's clear.

Reply:
> it wouldn't be if it was called with BindingMode::EXTRACT_NAMES

Exactly, the current implementation does not escape the table name, and I would not change the existing behaviour. But I agree that we should document this better - I'll add comments. 👍